### PR TITLE
SmartContract: Moved merkletree proof to it's own function

### PR DESCRIPTION
**Pull request by hackaugusto**
Small improvement on the netting channel code (#500)

### CODE REVIEW

File: raiden/smart_contracts/NettingChannelLibrary.sol [hunk 1]
```diff
@@ -300,32 +300,54 @@ library NettingChannelLibrary {
...
+        if (partner.locksroot != h) {
+            throw;
+        }
+
+        partner.unlocked.push(Lock(expiration, amount, hashlock));
+        self.locks[hashlock] = true;
+    }
 
-        if (partner.nonce == 0) {
+    function computeMerkleRoot(bytes lock, bytes merkle_proof)
```
Comment (czepluch) [inline raiden/smart_contracts/NettingChannelLibrary.sol]: Maybe make this function private since it's only used internally in the contract.

Comment (LefterisJP) [inline raiden/smart_contracts/NettingChannelLibrary.sol]: I think `internal` may be a better choice for functions like these.
Reference: http://solidity.readthedocs.io/en/develop/contracts.html?highlight=internal#visibility-and-getters

Comment (czepluch) [inline raiden/smart_contracts/NettingChannelLibrary.sol]: But also make it constant

**Comment by LefterisJP** [comment]
Closed since it's superseded by #503

**Pull request closed without merging**